sale_salesperson_filter/views/sale_order_view.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//group[@name='sales_person']/field[@name='user_id']" position="replace">
+                <field name="user_id" domain="[('salesperson_select','=', True)]" require="True"/>
```

Comment:
Pay attention to this: https://github.com/rfhk/cip-custom/blob/9970531a35063a6e50664a5d16b17ec17ffee241/sale_view_adjust/views/sale_order_views.xml#L39-L48

Also this https://github.com/rfhk/cip-custom/blob/87e965f7bab375f8d587995077e6299195d306fc/account_view_adjust/views/account_invoice_views.xml#L4-L13 when you work on invoice form view.